dynamic_instrumentation/src/point/build.rs

```diff
@@ -139,6 +139,36 @@ impl<'tcx> InstrumentationBuilder<'_, 'tcx> {
         self
     }
 
+    pub fn debug_mir(mut self, loc: Location) -> Self {
+        self.point.metadata.debug = {
+            let block = &self.body.basic_blocks()[loc.block];
+            if loc.statement_index == block.statements.len() {
+                match &block.terminator().kind {
+                    TerminatorKind::Call {
+                        args,
+                        destination,
+                        func,
+                        ..
+                    } if destination.is_some() => {
+                        let mut s = format!("{:?} = {:?}(", destination.unwrap().0, func);
+                        for (i, arg) in args.iter().enumerate() {
+                            if i > 0 {
+                                s.push_str(", ");
+                            }
+                            s.push_str(&format!("{:?}", arg));
+                        }
+                        s.push(')');
+                        s
+                    }
+                    _ => String::from(""),
+                }
+            } else {
+                format!("{:?}", block.statements[loc.statement_index])
+            }
```

Comment:
I don't like the unnecessary `destination.unwrap()`, though I get why it;s convenient here with the `match` `if`.  Maybe there's a better way to refactor this.

Reply:
Fixed in https://github.com/immunant/c2rust/pull/566/commits/cc477cbb94a1fb09a4d8f3d1f3412b3ae35f3727.